### ISSUE TITLE
Ford S-MAX connection

### ISSUE THREAD

**Issue by radimsejk** [meatpiHQ/wican-fw#286]
Hi, 

I struggle to understand data from the WICAN out of my car. I have the WICAN connected to car and paired to my WiFi. Mqtt connection also works. But:

1. I do not understand those incoming messages
2. I would like to extract tank level, coolant or some other engine temperature and battery voltage. 
3. Optional: I would like to extract error codes
4. Optional: I would like to extract state of my high beam lights (Blink high beam to do some action at home)

I have Ford S-Max 2012.

Here is the monitor screen:
![IMG_20241217_074609](https://github.com/user-attachments/assets/5213d9e2-1756-4ada-9bae-20013849a222)

Is there some manual or can someone help me with understanding those data? Thank you.

**Comment by meatpiHQ** [comment]
@radimsejk Please try the new Beta version, this is still not yet released. It will make it easier for you. After you update, set the protocol to AutoPid, and reboot. Then go to the Automate page, enable standard PID, select the ECU protocol and then scan for the available parameters. Let me know if you find any issues.

Make sure the unzip and upload the extracted bin file.
[wican-fw_obd_v400Beta3.zip](https://github.com/user-attachments/files/18161826/wican-fw_obd_v400Beta3.zip)

**Comment by radimsejk** [comment]
So after update I am not able to connect to my WiFi. When I connect to the Wican AP I can see all WiFi and MQTT settings is correct. But even after reboot it does not connect.

**Comment by meatpiHQ** [comment]
@radimsejk is your wifi network still using WPA2 only?

**Comment by radimsejk** [comment]
Yes. I have not changed it from last time. Could it be that the stored settings is somehow corrupted? Is there a way to "factory reset" the device?

But at least even without WiFi I can see some car variables now.
![IMG_20241218_072545](https://github.com/user-attachments/assets/c30a5606-520a-4d44-85b7-efd792db1b50)

**Comment by meatpiHQ** [comment]
@radimsejk no the stored settings are not corrupted. In this version I enabled WPA3 with WPA2 fallback. But it seems that the fallback is not working. You should consider changing your network to WPA3. 

I can compile a version with an option to select WPA2, but it's best to change to WPA3, since WPA2 is verunable.

**Comment by radimsejk** [comment]
The selector for WPA would be great. I think a lot of people is still using WPA2. I will try to run the WPA3 on my network but I am afraid of problems with my other devices. I heard that it could cause problems, maybe just like the Wican fallback bug.

**Comment by radimsejk** [comment]
I have verified that it works with WPA3 (tested using the hotspot on my phone). However, my home network does not support WPA3. I would greatly appreciate it if you could compile a WPA2 version or a version with a WPA selector. Thank you!

**Comment by meatpiHQ** [comment]
@radimsejk please try and let me know if it works for you.

[wican-fw_obd_v400Beta4.zip](https://github.com/user-attachments/files/18221213/wican-fw_obd_v400Beta4.zip)

**Comment by radimsejk** [comment]
"Should OTA be working? I’m unable to update from Beta 3 to Beta 4 using OTA. It says 'Updating,' but afterward, the page becomes unreachable. I have to reboot the device to get it working again, but it still runs Beta 3."

**Comment by radimsejk** [comment]
I tried updating via OTA using my PC (previously, I was using my phone), and the update was successful. I can now connect to my home network with WPA2 selected. Thank you again for your help! I will proceed with the integration and report back.

**Comment by radimsejk** [comment]
Finally, I got around to testing the beta firmware after the holidays. Connecting via WPA2 is stable, and connecting via MQTT to Node-RED also works. Using PID autodetection, I managed to add parameters that interest me. So far, everything is great.

However, I would now like to read these parameters in Node-RED via MQTT, but the only MQTT message contains the connection status and nothing else. Do I need to enable the sending of detected parameters via MQTT somewhere?

**Comment by meatpiHQ** [comment]
@radimsejk its not beta any more :) you should update to the latest firmware 

https://github.com/meatpiHQ/wican-fw/releases/tag/v4.03

> but the only MQTT message contains the connection status and nothing else.

This means that there is something wrong, is the ignition ON? if you are using Home assistant you should try the integration

https://meatpihq.github.io/wican-fw/home-assistant/integration_setup

**Comment by radimsejk** [comment]
Yes, the ignition is on, and the engine is running. I'll try installing the latest firmware and integration into Home Assistant. We'll see.

**Comment by radimsejk** [comment]
OK, I have new version installed and the HA integration added and I am now able to see those parameters there. Thank you very much for your help.
